app/jobs/change_pickup_library_job.rb

```diff
@@ -16,19 +16,30 @@ def perform(hold_key:, session_token:, pickup_library:)
     when 200
       human_readable_location = Hold::PICKUP_LOCATION_ACTUAL[pickup_library.to_sym]
       Redis.current.set("pickup_library_#{hold_key}", {
-        hold_id: hold_key,
+        id: hold_key,
         result: :success,
-        new_value: human_readable_location,
-        new_value_id: pickup_library
+        response: {
+          new_value: human_readable_location,
+          new_value_id: pickup_library
+        }
       }.to_json)
     else
-      error_message_raw = JSON.parse response.body
-      error_message = error_message_raw&.dig('messageList')&.first&.dig('message') || 'Something went wrong'
-      Sidekiq.logger.error("pickup_library_#{hold_key}: #{error_message}")
+      processed_error = SirsiResponse::Error.new(error_message_raw: JSON.parse(response.body),
```

Comment:
Wondering if we can DRY these jobs by adding a concern or some other way?

Reply:
Yeah. I guess the frustrating part for me is that it is DRYed out a bit. This was a stack of procedures being executed in order on every job and I replaced it with a `SirsiResponse::Error`. It just has a ton of arguments, which is why I think CC is complaining. There's probably a clever way to send less args. 🤷

Reply:
I was kinda thinking about some other abstractions that I ultimately left out. Like a model for the values we set in Redis instead of using a hash in a consistent manner. Thought about a lookup_title kind of utility module that could be mixed in with any job. Like this was the module:

```rb
module SirsiResponse::LookupTitle
  def get_title(symphony_client:, bib_type:, session_token:, identifier:)
    response = if bib_type == :checkout
                 SymphonyClientParser::parsed_response(symphony_client,
                                                       :get_item_info,
                                                       nil, session_token, identifier)
               else
                 SymphonyClientParser::parsed_response(symphony_client,
                                                       :get_hold_info,
                                                       key, session_token)
               end

    response.title
  end
end
```

Maybe that's better I don't know. I think I just went with this Error class instead because it is responsible for formatting and setting errors (which includes getting the title).

Reply:
> Yeah. I guess the frustrating part for me is that it is DRYed out a bit. This was a stack of procedures being executed in order on every job and I replaced it with a `SirsiResponse::Error`. It just has a ton of arguments, which is why I think CC is complaining. There's probably a clever way to send less args. 🤷

yes there are def. great improvements you implemented here 👌

Reply:
> I was kinda thinking about some other abstractions that I ultimately left out. Like a model for the values we set in Redis instead of using a hash in a consistent manner. Thought about a lookup_title kind of utility module that could be mixed in with any job. Like this was the module:
> 
> Maybe that's better I don't know. I think I just went with this Error class instead because it is responsible for setting errors (which includes getting the title).

Error class makes so much sense to me ✔️  As you mentioned just finding a way to send lesser arg.s might do make CC happy 🤔

Reply:
Would be great if the symph client instance and the token were globally accessible somehow, or modularized as a mixin to be included whereever needed. Just not sure it would work out.